modules/data-streams/src/javaRestTest/java/org/elasticsearch/datastreams/LogsIndexModeSettingRestTestIT.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.datastreams;
+
+import org.elasticsearch.Build;
+import org.elasticsearch.client.Request;
+import org.elasticsearch.client.Response;
+import org.elasticsearch.client.ResponseException;
+import org.elasticsearch.client.RestClient;
+import org.elasticsearch.index.IndexMode;
+import org.elasticsearch.test.cluster.ElasticsearchCluster;
+import org.elasticsearch.test.rest.ESRestTestCase;
+import org.junit.Before;
+import org.junit.ClassRule;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+import static org.hamcrest.Matchers.equalTo;
+
+public class LogsIndexModeSettingRestTestIT extends ESRestTestCase {
+
+    @ClassRule
+    public static ElasticsearchCluster cluster = ElasticsearchCluster.local()
+        .module("x-pack-stack")
+        .module("data-streams")
+        .setting("stack.templates.enabled", "true")
```

Comment:
This should not really be required since the default is "true"

Reply:
Right, but maybe this is disabled somewhere else in this test cluster setup? It defaults to true, so something must be setting it to false? What do you see when you debug `StackPlugin`? Is this plugin loaded btw?

Reply:
If I debug the test I don't see it going into `StackTemplateRegistry` constructor where the setting is evaluated...so I am not sure that is the issue. I think something is missing to actually use the registry.